haystack/nodes/prompt/invocation_layer/open_ai.py

```diff
@@ -270,7 +272,7 @@ def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union
 
     @classmethod
     def supports(cls, model_name_or_path: str, **kwargs) -> bool:
-        valid_model = model_name_or_path in ["ada", "babbage", "davinci", "curie", "gpt-3.5-turbo-instruct"] or any(
+        valid_model = model_name_or_path in OPEN_AI_MODEL_NAMES or any(
```

Comment:
Does this work for the snapshot models (e.g. `gpt-3.5-turbo-0613`)?

Reply:
No :/ Basically we need to come up with a rule to match all OpenAI models or somehow with a hard coded rule or implement some kind of dynamic validation by e.g. fetching all available models and check if the chosen one is one of them. 

Happy to add extra model names